service.sleepy.watchdog/default.py

```diff
@@ -0,0 +1,293 @@
+        notifyLog('Action:                   %s' % (self.action))
+        notifyLog('Jump to main menue:       %s' % (self.jumpMainMenu))
+        notifyLog('keep alive:               %s' % (self.keepAlive))
+        notifyLog('run addon:                %s' % (self.addon_id))
+
+        # following commented code are preparations for actions within an alternate timeframe,
+        # don't shure if I ever implement this...
+        """
+        notifyLog('use alternate time frame: %s' % (self.use_alternate_frame))
+        notifyLog('alt. max. idle time:      %s' % (self.alternate_maxIdleTime))
+        notifyLog('alternate action:         %s' % (self.alternate_action))
+        notifyLog('Jump to main menu:        %s' % (self.alternate_jumpMainMenu))
+        notifyLog('keep alive:               %s' % (self.alternate_keepAlive))
+        notifyLog('run addon:                %s' % (self.alternate_addon_id))
+        notifyLog('Test run:                 %s' % (self.testConfig))
+        """
+
+        if self.testConfig:
+            self.maxIdleTime = 60 + int(self.notifyUser)*self.notificationTime
+            notifyLog('running in test mode for %s secs' % (self.maxIdleTime))
+
+    def activeTimeFrame(self, debug=False):
+
+        if not self.timeframe: return True
+
+        _currframe = int((datetime.datetime.now() - datetime.datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)).total_seconds())
+
+        if _currframe < self.currframe: _currframe += 86400
+        self.currframe = _currframe
+
+        if debug: notifyLog('checking time frames: start: %s - current: %s - end: %s' % (self.act_start, self.currframe, self.act_stop))
+        if self.act_start <= self.currframe <= self.act_stop: return True
+        return False
+
+    # user defined actions
+
+    def stopVideoAudioTV(self):
+        if xbmc.Player().isPlaying():
+            notifyLog('media is playing, stopping it')
+            xbmc.Player().stop()
+            if self.jumpMainMenu:
+                xbmc.sleep(500)
+                notifyLog('jump to main menu')
+                xbmc.executebuiltin('ActivateWindow(home)')
+
+    @classmethod
+    def quit(cls):
+        notifyLog('quit kodi')
+        xbmc.executebuiltin('Quit')
+
+    @classmethod
+    def systemReboot(cls):
+        notifyLog('init system reboot')
+        xbmc.restart()
+
+    @classmethod
+    def systemShutdown(cls):
+        notifyLog('init system shutdown')
+        xbmc.shutdown()
+
+    @classmethod
+    def systemHibernate(cls):
+        notifyLog('init system hibernate')
+        xbmc.executebuiltin('Hibernate')
+
+    @classmethod
+    def systemSuspend(cls):
+        notifyLog('init system suspend')
+        xbmc.executebuiltin('Suspend')
+
+    def sendCecCommand(self):
+        if not self.sendCEC: return
+        notifyLog('send standby command via CEC')
+        cec = subprocess.Popen('echo "standby 0" | cec-client -s', stdout=subprocess.PIPE, shell=True).communicate()
```

Comment:
That'll work only on unix systems right?
I'm asking because the addon itself says platform -> all